### PR TITLE
Rename to SSO Kit

### PR DESCRIPTION
**Pull request by heruan**
This performs the following renaming:

- `auth` -> `sso-kit`
- `auth-start` -> `sso-kit-starter`
- `auth-demo` -> `sso-kit-demo`
- `com.vaadin.auth.*` -> `com.vaadin.sso.*`
- `VaadinAuthSecurityConfiguration` -> `SingleSignOnConfiguration`
- Spring properties `vaadin.auth.*` -> `vaadin.sso.*`

and other minor classes accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -110,4 +117,4 @@
         </plugins>
     </build>
 
-</project>
+</project>
```
Comment (DiegoCardoso) [inline pom.xml:120]: nit: restore newline

File: sso-kit-starter/src/main/java/com/vaadin/sso/starter/SingleSignOnConfiguration.java [hunk 2]
```diff
@@ -145,10 +145,10 @@ protected void configure(HttpSecurity http) throws Exception {
         });
     }
 
-    static class DefaultVaadinAuthContext implements VaadinAuthContext {
+    static class DefaultVaadinAuthContext implements AuthenticationContext {
```
Comment (DiegoCardoso) [inline sso-kit-starter/src/main/java/com/vaadin/sso/starter/SingleSignOnConfiguration.java:148]: Shouldn't the class be also renamed to something like `DefaultAuthenticationContext`?
Same thing would apply to the property name defined on line `61`.

File: sso-kit-starter/src/main/resources/META-INF/spring.factories [hunk 3]
```diff
@@ -1,2 +1,2 @@
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
-com.vaadin.auth.starter.VaadinAuthSecurityConfiguration
\ No newline at end of file
+com.vaadin.sso.starter.SsoKitConfiguration
```
Comment (DiegoCardoso) [inline sso-kit-starter/src/main/resources/META-INF/spring.factories:2]: nit: add newline at the end of the file

**Review by DiegoCardoso — CHANGES_REQUESTED** [review]
Apart from the minor changes that I pointed out, another "nit" is that the theme on the demo project still references to `auth-demo`.

**Comment by heruan** [comment]
> Apart from the minor changes that I pointed out, another "nit" is that the theme on the demo project still references to `auth-demo`.

Right! I've renamed also the theme now.

**Review by DiegoCardoso — APPROVED** [review]

**Comment by jouni** [comment]
Rename the repo to sso-kit as well?

**Comment by heruan** [comment]
> Rename this repo to sso-kit as well?

Yes, it's planned! It requires some changes to CI/CD so we'll do presumably next week.

**Pull request merged**